src/nvim/api/buffer.c

```diff
@@ -453,20 +486,18 @@ Object nvim_buf_set_var(Buffer buffer, String name, Object value, Error *err)
 /// @param name The variable name
 /// @param[out] err Details of an error that may have occurred
 /// @return The old value or nil if there was no previous value.
-///
-///         @warning It may return nil if there was no previous value
-///                  or if previous value was `v:null`.
-Object nvim_buf_del_var(Buffer buffer, String name, Error *err)
+Object buffer_del_var(Buffer buffer, String name, Error *err)
```

Comment:
No, it is only the old value, if key is not found it is an error. Thanks for noting!